src/sqlfluff/core/dialects/__init__.py

```diff
@@ -81,7 +81,7 @@ def dialect_readout() -> Iterator[DialectTuple]:
 
 def dialect_selector(s: str) -> Dialect:
     """Return a dialect given its name."""
-    dialect = load_raw_dialect(s or "ansi")
+    dialect = load_raw_dialect(s)
```

Comment:
Here, we no longer hardcode `"ansi"` as a fallback if no dialect was provided. Let it fail if it gets this far.